module/GenotypeCalls.py

```diff
@@ -374,6 +374,20 @@ def get_base_calls_forward_strand(self, snps, forward_strand_annotations):
         """
         return self.get_base_calls_generic(snps, forward_strand_annotations, SourceStrand.Forward, RefStrand.Unknown)
 
+    def get_base_calls_TOP_strand(self, snps, ilmn_strand_annotations):
+        """
+        Get base calls on the Illumina strand.
+
+        Args:
+            snps (list(string)) : A list of string representing the snp on the design strand for the loci (e.g. [A/C])
+            ilmn_strand_annotations (list(int)) : A list of strand annotations for the loci (e.g., IlmnStrand.TOP)
+
+        Returns:
+            The genotype basecalls on the report strand as a list of strings.
+            The characters are A, C, G, T, or - for a no-call/null.
+        """
+        return self.get_base_calls_generic(snps, ilmn_strand_annotations, IlmnStrand.TOP, RefStrand.Unknown)
```

Comment:
e.g., here could potentially pass
return self.get_base_calls_generic(snps, ilmn_strand_annotations, [IlmnStrand.TOP, IlmnStrand.PLUS], IlmnStrand.Unknown)

Reply:
Also, should be passing IlmnStrand.Unknown here instead of RefStrand.Unknown. (get_base_calls_forward strand also passes RefStrand.Unknown, but that is probably not the correct behavior there either)

Reply:
Added list support for get_base_calls_generic. That should solve the ambiguity. The list type for the method is optional in order to leave the other methods untouched. Within the method converting non list types to list before iterating through every allele as list type checks in the inner loop might be something of a performance drain compared to looping through only one element. 

Changed to IlmnStrand.Unknown.